app/models/miq_worker/container_common.rb

```diff
@@ -44,15 +44,20 @@ def default_image
     def resource_constraints
       mem_threshold = self.class.worker_settings[:memory_threshold]
       cpu_threshold = self.class.worker_settings[:cpu_threshold_percent]
+      mem_default   = self.class.worker_settings[:memory_minimum]
+      cpu_default   = self.class.worker_settings[:cpu_minimum_percent]
 
-      return {} if !Settings.server.worker_monitor.enforce_resource_constraints || (mem_threshold.nil? && cpu_threshold.nil?)
+      return {} unless Settings.server.worker_monitor.enforce_resource_constraints
+      return {} unless mem_threshold || cpu_threshold || mem_default || cpu_default
```

Comment:
I think this line can just go away... the method will still return an empty {} with the code you have below.